leshan-tl-cf-server-coap/src/test/java/org/eclipse/leshan/transport/californium/server/observation/ObservationServiceTest.java

```diff
@@ -262,4 +265,21 @@ public void stop() {
         public void destroy() {
         }
     }
+
+    private class ExtendedObservation extends Observation {
+        public ExtendedObservation(ObservationIdentifier id, String registrationId, Map<String, String> context,
+                Map<String, String> protocolData) {
+            super(id, registrationId, context, protocolData);
+        }
+
+        @Override
+        public boolean canEqual(Object obj) {
+            return (obj instanceof ExtendedObservation);
+        }
+    }
+
+    @Test
+    public void assertEqualsHashcode() {
+        EqualsVerifier.forClass(Observation.class).withRedefinedSubclass(ExtendedObservation.class).verify();
```

Comment:
OK, I used SingleObservation

Reply:
There are 2 - SingleObservation & CompositeObservation but for now I'm facing a problem with integration tests which involves them

Reply:
I found a reason - it was because of **Observation** class. I added **protocolData** field to rewritten equals&hashCode and it wasn't there at the previous, old implementation. I guess that it was ommited on purpose and it should stay that way?

Reply:
Yep correct.